packages/account/src/Sections/Verification/ProofOfIdentity/proof-of-identity-container.jsx

```diff
@@ -18,220 +18,245 @@ import Verified from 'Components/poi/status/verified';
+        const handleRequireSubmission = () => setHasRequireSubmission(true);
+        const country_code = account_settings?.citizen || account_settings?.country_code;
+
+        const handleManualSubmit = () => {
+            WS.authorized.getAccountStatus().then(() => {
+                refreshNotifications();
+            });
+        };
+        const loadResidenceList = React.useCallback(() => {
+            setAPIError(null);
+            setStatusLoading(true);
+            fetchResidenceList().then(response_residence_list => {
+                if (response_residence_list.error) {
+                    setAPIError(response_residence_list.error);
+                } else {
+                    setResidenceList(response_residence_list.residence_list);
                 }
-                loadResidenceList();
             });
-        }
-    }, [is_switching, loadResidenceList]);
+            setStatusLoading(false);
+        }, [fetchResidenceList]);
 
-    if (api_error) {
-        return <ErrorMessage error_message={api_error?.message || api_error} />;
-    }
-    /**
-     * Display loader while waiting for the account status and residence list to be populated
-     */
-    if (is_status_loading || is_switching || isEmptyObject(account_status) || residence_list.length === 0) {
-        return <Loading is_fullscreen={false} />;
-    } else if (is_virtual) {
-        return <DemoMessage />;
-    }
+        React.useEffect(() => {
+            if (is_language_changing) {
+                loadResidenceList();
+            }
+        }, [is_language_changing, loadResidenceList]);
 
-    const verification_status = populateVerificationStatus(account_status);
-    const {
-        idv,
-        allow_poi_resubmission,
-        identity_last_attempt,
-        identity_status,
-        is_age_verified,
-        is_idv_disallowed,
-        manual,
-        needs_poa,
-        onfido,
-    } = verification_status;
-    const should_ignore_idv = is_high_risk && is_withdrawal_lock;
-
-    if (!should_allow_authentication && !is_age_verified) {
-        return <NotRequired />;
-    }
+        React.useEffect(() => {
+            // only re-mount logic when switching is done
+            if (!is_switching) {
+                setAPIError(null);
+                WS.authorized.getAccountStatus().then(response_account_status => {
+                    if (response_account_status.error) {
+                        setAPIError(response_account_status.error);
+                        setStatusLoading(false);
+                        return;
+                    }
+
+                    loadResidenceList();
+                });
+            }
+        }, [loadResidenceList, is_switching]);
+        /**
+         * Display loader while waiting for the account status and residence list to be populated
+         */
+        if (is_status_loading || is_switching || isEmptyObject(account_status)) {
```

Comment:
is there any reason we remove this condition (residence_list.length === 0) from previous implementation in this if condition

Reply:
It is a mistake. Code is overwritten